go/multipooler/manager/manager.go

```diff
@@ -394,6 +394,36 @@ func (pm *MultiPoolerManager) getPoolerType() clustermetadatapb.PoolerType {
 	return clustermetadatapb.PoolerType_UNKNOWN
 }
 
+// getDatabase returns the database name from the multipooler record
+func (pm *MultiPoolerManager) getDatabase() string {
+	pm.cachedMultipooler.mu.Lock()
+	defer pm.cachedMultipooler.mu.Unlock()
+	if pm.cachedMultipooler.multipooler != nil && pm.cachedMultipooler.multipooler.MultiPooler != nil {
+		return pm.cachedMultipooler.multipooler.Database
+	}
+	return ""
+}
+
+// getBackupLocation returns the backup location from the database topology
+func (pm *MultiPoolerManager) getBackupLocation(ctx context.Context) (string, error) {
+	database := pm.getDatabase()
+	if database == "" {
+		return "", mterrors.New(mtrpcpb.Code_FAILED_PRECONDITION, "database name not set in multipooler")
+	}
+
+	db, err := pm.topoClient.GetDatabase(ctx, database)
+	if err != nil {
+		return "", mterrors.Wrapf(err, "failed to get database %s from topology", database)
+	}
+
+	if db.BackupLocation == "" {
```

Comment:
One benefit on my idea above, is that we can make sure that any errors reading this information, could be used to make sure the pooler never transitions to a ready instead if this information is not available.